keps/140-scenario-based-simulation/README.md

```diff
@@ -43,65 +64,85 @@ Therefore, this kep proposes to define the scenario **as CRD**. All clients, inc
 The Scenario is a non-namespaced resource. 
 This CRD will be applied to kube-apiserver started in kube-scheduler-simulator.
 
-We may need to change etcd request-size limitation by --max-request-bytes since the scenario resource may be bigger than other normal resources.
+We may need to change etcd request-size limitation by --max-request-bytes since the scenario resource may be more significant than other standard resources.
 https://etcd.io/docs/v3.4/dev-guide/limit/#request-size-limit
 
 ```go
-// Scenario is the Schema for the scenarios API
-type Scenario struct {
-  metav1.TypeMeta   `json:",inline"`
-  metav1.ObjectMeta `json:"metadata,omitempty"`
-
-  Spec   ScenarioSpec   `json:"spec,omitempty"`
-  Status ScenarioStatus `json:"status,omitempty"`
-}
-
 // ScenarioSpec defines the desired state of Scenario
 type ScenarioSpec struct {
-	// Events field has all operations for a scenario.
-	// Also you can add new events during the scenario is running.
+	// Operations field has all operations for a scenario.
+	// Also, you can add a new operation while the scenario runs.
 	//
 	// +patchMergeKey=ID
 	// +patchStrategy=merge
-	Events []*ScenarioEvent `json:"events"`
+	Operations []*ScenarioOperation `json:"operations"`
+
+	// Controllers have the configuration for controllers working with simulation.
+	Controllers *Controllers `json:"controllers"`
+}
+
+type Controllers struct {
+	// PreparingControllers is a list of controllers that should be run before SimulatedControllers.
+	// They will run in parallel.
+	// 
+	// It's an optional field. 
+	// All controllers registered in the simulator will be enabled automatically. (except controllers set in Simulate.)
+	// So, you need to configure it only when you want to disable some controllers enabled by default.
+	//
+	// +optional
+	PreparingControllers    *ControllerSet `json:"preparingControllers`
+	// SimulatedControllers is a list of controllers that are the target of this simulation.
+	// These are run one by one in the same order specified in Enabled field.
+	// 
+	// It's a required field; no controllers will be enabled automatically.
+	SimulatedControllers  *ControllerSet `json:"simulatedControllers"`
 }
 
-type ScenarioEvent struct {
-	// ID for this event. Normally, the system sets this field for you.
+type ControllerSet struct {
+	// Enabled specifies controllers that should be enabled.
+	// +listType=atomic
+	Enabled  []Controller `json:"enabled"`
+	// Disabled specifies controllers that should be disabled.
+	// When all controllers need to be disabled, an array containing only one "*" should be provided.
+	// +listType=map
+	// +listMapKey=name
+	Disabled []Controller `json:"disabled"`
```

Comment:
What is this mean if we register a controller to `Disabled` of SimulatedControllers. Is there some effect?